pyannote/audio/core/model.py

```diff
@@ -252,13 +259,15 @@ def setup(self, stage=None):
                 module.to(self.device)
 
         # add (trainable) loss function (e.g. ArcFace has its own set of trainable weights)
-        if stage == "fit":
+        if self.task:
             # let task know about the model
             self.task.model = self
-            # setup custom loss function
-            self.task.setup_loss_func()
-            # setup custom validation metrics
-            self.task.setup_validation_metric()
+            # setup loss function and validation metric on training and validation only
+            if stage in ["fit", "validate"]:
```

Comment:
Isn't this redundant with `if self.task`? 
If not, can you please clarify with a comment why this is needed?

Reply:
> isn't this redundant with if self.task?

I think so